src/NDItk/depletion/ReactionMultiplicities/test/ReactionMultiplicities.test.cpp

```diff
@@ -0,0 +1,302 @@
+            "be tested" ) {
+
+        verifyChunkWithMultiplicityType( chunk );
+      } // THEN
+
+      THEN( "the record can be printed" ) {
+
+        std::string buffer;
+        auto output = std::back_inserter( buffer );
+        chunk.print( output );
+
+        CHECK( buffer == record );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+
+  GIVEN( "invalid data for a ReactionMultiplicities instance" ) {
+
+    WHEN( "the number of reactions is insufficient" ) {
+
+      std::vector< Multiplicities > multiplicities = {};
+
+      THEN( "an exception is thrown" ) {
+
+        CHECK_THROWS( ReactionMultiplicities( std::move( multiplicities ) ) );
+      } // THEN
+    } // WHEN
+
+    WHEN( "reading the data of the record and the number of reactions "
+          "values is insufficient" ) {
+
+      std::string record = chunkWithInsufficientNumberReactions();
+      auto iter = record.begin() + 5;
+      auto end = record.end();
+      ReactionMultiplicities chunk;
+
+      THEN( "an exception is thrown" ) {
+
+        CHECK_THROWS( chunk.read( iter, end, 0 ) );
+      } // THEN
+    } // WHEN
+  } // GIVEN
+} // SCENARIO
+
+std::string chunk() {
+
+  return "rprod\n"
+         "    2\n"
+         "    2\n"
+         "    1 1\n"
+         "    92235 1\n"
+         "    16\n"
+         "    2\n"
+         "    1 2\n"
+         "    92234 1\n";
+}
+
+void verifyChunk( const ReactionMultiplicities& chunk ) {
+
+  CHECK( "rprod" == chunk.keyword() );
+  CHECK(     2 == chunk.values()[0] );
+  CHECK(     2 == chunk.values()[1] );
+  CHECK(     1 == chunk.values()[2] );
+  CHECK(     1 == chunk.values()[3] );
+  CHECK( 92235 == chunk.values()[4] );
+  CHECK(     1 == chunk.values()[5] );
+  CHECK(    16 == chunk.values()[6] );
+  CHECK(     2 == chunk.values()[7] );
+  CHECK(     1 == chunk.values()[8] );
+  CHECK(     2 == chunk.values()[9] );
+  CHECK( 92234 == chunk.values()[10] );
+  CHECK(     1 == chunk.values()[11] );
+
+  CHECK( false == chunk.empty() );
+  CHECK( 12 == chunk.size() );
+
+  CHECK( 2 == chunk.numberReactions() );
+
+  CHECK( true == chunk.hasReaction( 2 ) );
+  CHECK( true == chunk.hasReaction( 16 ) );
+  CHECK( false == chunk.hasReaction( 102 ) );
+
+  CHECK( 2 == chunk.reactions()[0].identifier() );
+  CHECK( 2 == chunk.reactions()[0].numberReactionProducts() );
+  CHECK( 1 == chunk.reactions()[0].reactionProducts()[0] );
+  CHECK( 92235 == chunk.reactions()[0].reactionProducts()[1] );
+  CHECK( 1 == chunk.reactions()[0].multiplicities()[0] );
+  CHECK( 1 == chunk.reactions()[0].multiplicities()[1] );
+
+  CHECK( 16 == chunk.reactions()[1].identifier() );
+  CHECK( 2 == chunk.reactions()[1].numberReactionProducts() );
+  CHECK( 1 == chunk.reactions()[1].reactionProducts()[0] );
+  CHECK( 92234 == chunk.reactions()[1].reactionProducts()[1] );
+  CHECK( 2 == chunk.reactions()[1].multiplicities()[0] );
+  CHECK( 1 == chunk.reactions()[1].multiplicities()[1] );
+
+  auto multiplicities = chunk.reaction( 2 );
```

Comment:
Did you mean to use multiplicities rather than chunk.reaction( 2 ) below this?

Reply:
I removed the line in fix/review-part6